### PR TITLE
Add SparkJobUploadArtifactException when uploading file by blob with wrong input

### PR DESCRIPTION
**Pull request by lcadzy**
#2580

### CODE REVIEW

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkJobUploadArtifactException.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+package com.microsoft.azure.hdinsight.spark.common;
+
+public class SparkJobUploadArtifactException extends SparkJobException {
+    public SparkJobUploadArtifactException(String message, Throwable throwable) {
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkJobUploadArtifactException.java]:
```suggestion
    public SparkJobUploadArtifactException(String message, Throwable cause) {
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/common/classifiedexception/SparkUserException.kt [hunk 2]
```diff
@@ -35,7 +35,8 @@ object SparkUserExceptionFactory : ClassifiedExceptionFactory() {
     override fun createClassifiedException(exp: Throwable?): ClassifiedException? {
         return if (exp is YarnDiagnosticsException
                 // Throw from wrong class name ,refer to issue 1827 and 2466
-                || exp is SparkJobException) {
+                || exp is SparkJobException
+                || exp is IllegalArgumentException) {
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/common/classifiedexception/SparkUserException.kt]: Not handle `IllegalSecuryKeyException`?

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobRemoteProcess.java [hunk 3]
```diff
@@ -222,6 +223,8 @@ protected void ctrlError(String message) {
     protected Observable<? extends ISparkBatchJob> prepareArtifact() {
         return getSparkJob()
                 .deploy(artifactPath)
+                .doOnError(err ->
+                   Observable.error(new SparkJobUploadArtifactException("Upload artifact encounters exception " + err.getMessage(), err)))
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobRemoteProcess.java]:
```suggestion
                   Observable.error(new SparkJobUploadArtifactException("Failed to upload Spark application artifacts: " + err.getMessage(), err)))
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/common/classifiedexception/SparkUserException.kt (hunk 2 above)
Comment (lcadzy) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/common/classifiedexception/SparkUserException.kt]: if key is invalid ,the container is null which handles in running configuration

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/jobs/JobUtils.java [hunk 4]
```diff
@@ -443,30 +444,30 @@ private static BlobContainer getSparkClusterDefaultContainer(ClientStorageAccoun
...
         if(storageAccount.getAccountType() == StorageAccountTypeEnum.BLOB) {
             try (FileInputStream fileInputStream = new FileInputStream(file)) {
                 try (BufferedInputStream bufferedInputStream = new BufferedInputStream(fileInputStream)) {
                     HDStorageAccount blobStorageAccount = (HDStorageAccount) storageAccount;
-                    BlobContainer defaultContainer = getSparkClusterDefaultContainer(blobStorageAccount, defaultContainerName);
-
-                    if (defaultContainer == null) {
-                        throw new UnsupportedOperationException("Can't get the default container.");
+                    BlobContainer container = getSparkClusterContainer(blobStorageAccount, containerName);
+
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/jobs/JobUtils.java]: Whitespace

**Comment by lcadzy** [comment]
#2653

**Pull request merged**